contracts/partial/dex_core/dex_core_lambdas.ligo

```diff
@@ -352,30 +360,26 @@ function swap(
           params.swaps,
           record [
             s               = s;
-            ops             = (nil : list(operation));
+            forwards        = (nil : list(forward_t));
             last_operation  = (None : option(operation));
             token_in        = token;
             receiver        = params.receiver;
             referrer        = params.referrer;
+            from_bucket     = Constants.zero_address;
             amount_in       = params.amount_in;
             swaps_list_size = List.size(params.swaps);
             counter         = 0n;
           ]
         );
+        const forward_ops : list(operation) = List.fold(create_pour_over_op, tmp.forwards, (nil : list(operation)));
 
         assert_with_error(tmp.amount_in >= params.min_amount_out, DexCore.err_high_min_out);
```

Comment:
There is the minor gas optimisation. The operations order can be changed: assert first and calculating forward_ops after. In case if the result amount out isn't sufficient, unnecessary computations won't be done.